app/screens/select_server/select_server.js

```diff
@@ -233,20 +233,33 @@ export default class SelectServer extends PureComponent {
     handleLoginOptions = async () => {
         const {formatMessage} = this.context.intl;
         const {config, license} = this.props;
-        const samlEnabled = config.EnableSaml === 'true' && license.IsLicensed === 'true' && license.SAML === 'true';
+        const isLicensed = license.IsLicensed === 'true';
+        const samlEnabled = config.EnableSaml === 'true' && isLicensed && license.SAML === 'true';
         const gitlabEnabled = config.EnableSignUpWithGitLab === 'true';
-        const googleEnabled = config.EnableSignUpWithGoogle === 'true' && license.IsLicensed === 'true';
-        const o365Enabled = config.EnableSignUpWithOffice365 === 'true' && license.IsLicensed === 'true' && license.Office365OAuth === 'true';
-        const openIdEnabled = config.EnableSignUpWithOpenId === 'true' && license.IsLicensed === 'true' && isMinimumServerVersion(config.Version, 5, 33, 0);
-
-        let options = 0;
-        if (samlEnabled || gitlabEnabled || googleEnabled || o365Enabled || openIdEnabled) {
-            options += 1;
-        }
+        const googleEnabled = config.EnableSignUpWithGoogle === 'true' && isLicensed;
+        const o365Enabled = config.EnableSignUpWithOffice365 === 'true' && isLicensed && license.Office365OAuth === 'true';
+        const openIdEnabled = config.EnableSignUpWithOpenId === 'true' && isLicensed === 'true';
```

Comment:
isLicensed is never equals to 'true', since it is a boolean.

Also, I would make all this conditions to have isLicensed as the first condition, mainly to make it more readable.